data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/Event.java

```diff
@@ -53,6 +53,30 @@ public interface Event extends Serializable {
      */
     <T> List<T> getList(String key, Class<T> clazz);
 
+    /**
+     * Retrieves the given key, of generic type Object, from the Event.
+     * Used by the Ruby processor to
+     * provide an Event API from within Ruby.
+     *
+     * @param key the value to retrieve from
+     * @param <T> The type
+     * @return T an Object from the key
+     * @since 2.4
+     */
+    <T> T get(String key);
+
+    /**
+     * Retrieves the given key, of generic type Object, from the Event as a List.
+     * Used by the Ruby processor to
+     * provide an Event API from within Ruby.
+     *
+     * @param key the value to retrieve from
+     * @param <T> The type
+     * @return {@literal List<T>} a list of Object elements
+     * @since 2.4
+     */
+    <T> List<T> getList(String key);
```

Comment:
I suggested Finn go this route to avoid requiring the second parameter in the ruby code. The logstash ruby plugin doesn't require the second parameter, so this helps acheive parity with logstash.

Reply:
Thanks for the feedback and coupling the Ruby processor and Event is also a concern of mine. 
Another issue with keeping the second parameter is that any Ruby code needs to do some complicated footwork to send the right class literal. So `get(key, Object.class)` would require something along the lines of `require 'java' \n get(key, java.lang.Object.class)` within Ruby. 
An alternative implementation that I spent some time thinking about was creating a specific RubyEvent for this purpose. Going down this route would require some type casting within the processor. If I recall it would also require some changes to Event like making some methods `protected` instead of `private`. 

What do you think about this solution, or keeping the original modification to `get` and `getList`?

Reply:
A third alternative is to have users write `get(field)` in their code and expand these calls to `get(field, java_object_class_literal)` before they are sent into the JRuby environment. While normally these string manipulation are fragile, the fact that there is only one expansion makes me more comfortable suggesting this.

Reply:
You can create a `RubyEvent` in the Ruby Processor project. 


```
class RubyEvent implements Event {
  RubyEvent(Event delegate) {
    this.delegate = delegate;
  }

  <T> T get(String key) {
    // your implementation
  }

  // follow this pattern for any existing methods that don't need different behavior.
  <T> T get(String key, Class<T> clazz) {
    return delegate.get(key, clazz);
  }
```